adapters/huaweiads/huaweiads.go

```diff
@@ -524,17 +560,31 @@ func getNativeFormat(adslot30 *adslot30, openRTBImp *openrtb2.Imp) error {
 	var detailedCreativeTypeList = make([]string, 0, 2)
 	if numVideo >= 1 {
 		detailedCreativeTypeList = append(detailedCreativeTypeList, "903")
-	} else if numImage > 1 {
+	} else if numMainImage > 1 {
 		detailedCreativeTypeList = append(detailedCreativeTypeList, "904")
-	} else if numImage == 1 {
-		detailedCreativeTypeList = append(detailedCreativeTypeList, "909")
+	} else if numMainImage == 1 {
+		detailedCreativeTypeList = append(detailedCreativeTypeList, "901")
 	} else {
 		detailedCreativeTypeList = append(detailedCreativeTypeList, "913", "914")
 	}
 	adslot30.DetailedCreativeTypeList = detailedCreativeTypeList
 	return nil
 }
 
+// roll ad need TotalDuration
+func getVideoFormat(adslot30 *adslot30, adtype int32, openRTBImp *openrtb2.Imp) error {
+	adslot30.W = openRTBImp.Video.W
+	adslot30.H = openRTBImp.Video.H
+
+	if adtype == roll {
+		if openRTBImp.Video.MaxDuration == 0 {
+			return errors.New("GetHuaweiAdsReqAdslot30: MaxDuration is empty when adtype is roll.")
+		}
+		adslot30.TotalDuration = int32(openRTBImp.Video.MaxDuration)
+	}
+	return nil
+}
+
 func convertAdtypeString2Integer(adtypeLower string) int32 {
```

Comment:
Could you rename this to `convertAdtypeStringToInteger()`

Reply:
changed